src/bindings/c/ov/include/c_api/ov_c_api.h

```diff
@@ -349,9 +366,17 @@ typedef struct ov_property{
 
 /**
  * @brief Initialize a partial shape.
+ * @param str is the input partial info string
+ *  Dynamic rank:
+ *     Example: "?"
+ *  Static rank, but dynamic dimensions on some or all axes.
+ *     Examples: "{1,2,?,4}" or "{?,?,?}" or "{1,2,-1,4}""
+ *  Static rank, and static dimensions on all axes.
+ *     Examples: "{1,2,3,4}" or "{6}" or "{}""
+ *
  * @param ov_status_e a status code.
  */
-OPENVINO_C_API(ov_status_e) ov_partial_shape_init(ov_partial_shape_t* partial_shape, const char* str);
+OPENVINO_C_API(ov_status_e) ov_partial_shape_init(ov_partial_shape_t** partial_shape, const char* str);
```

Comment:
Can we use ov::Dimension instead of string for initialization

Reply:
For C, it is hard to define a flexible structure for dynamic dimension, could you give some idea? Thanks!

Reply:
We can have the same structure as in C++, dimension contains interval (2 values) and use the same logic as in C++:
`m_dimension(dimension == -1 ? 0 : dimension, dimension == -1 ? Interval::s_max : dimension)`.

Reply:
Can it work for below scenario?

 *  Dynamic rank:
 *     Example: "?"
 *  Static rank, but dynamic dimensions on some or all axes.
 *     Examples: "{1,2,?,4}" or "{?,?,?}" or "{1,2,-1,4}""
 *  Static rank, and static dimensions on all axes.
 *     Examples: "{1,2,3,4}" or "{6}" or "{}""
 *

Anymore, dimension=0 is meaningful for some special model, https://jira.devtools.intel.com/browse/CVS-85804

Reply:
Rank in C++ is `ov::Dimension`: https://github.com/openvinotoolkit/openvino/blob/master/src/core/include/openvino/core/rank.hpp#L13

It means that if we use this structure for rank we can set dynamic rank to.

Dynamic dimension is empty dimension (`ov::Dymension()`) or dimension with interval (`ov::Dymension(min_val, max_val)`).

All your cases can be covered by using `ov::Dimension` structure for rank and dimensions.